bootable-jar/WFLY-13305_Bootable_Jar.adoc

```diff
@@ -0,0 +1,469 @@
+* The ```jboss.node.name``` system propery, if not set, is set to the value of ```HOSTNAME``` environment variable if defined (defined in OpenShift PODS). If HOSTNAME is not set 
+and jboss.node.name is not set, jboss.node.name is not set. The node name value is truncated to a max of 23 characters in order for the transaction subsystem to properly operate. The last 23 characters are kept in order to avoid conflicts.
+* The server logs are printed in the console.
+* jgroups subsystem is configured to use kubernetes.KUBE_PING jgroups protocol for both tcp (default stack) and udp. PING and MPING protocols are removed.
+* It is possible to configure jgroups to use un-encrypted password authentication. Set the ```<cloud>``` child element ```<enable-jgroups-password>true|false</enable-jgroups-password>``` to enable authentication. 
+NB: When authentication is enabled, the environment variable ```JGROUPS_CLUSTER_PASSWORD``` must be set otherwise the server will fail to start (the password expression being un-resolved).
+
+Some examples:
+
+Configure for OpenShift execution:
+
+```
+<cloud/>
+```
+
+Configure for OpenShift execution with jgroups authentication enabled:
+
+```
+<cloud>
+  <enable-jgroups-password>true</enable-jgroups-password>
+</cloud>
+```
+Configure for kubernetes execution:
+
+```
+<cloud>
+  <type>kubernetes</type>
+</cloud>
+```
+
+===== WildFly OpenShift operator
+
+The WildFly OpenShift operator can be used to manage deployments based on image containing a WildFly bootable JAR.
+At boot time, the WildFly bootable JAR dumps in the file ```/opt/jboss/container/wildfly-bootable-jar/install-dir``` its installation path.
+This information is required by the WildFly OpenShift operator to retrieve transaction logs and call into WildFly CLI.
+
+===== JKube Maven plugin
+
+The JKube Maven Plugin (version 1.0.0-rc-1 for now) has been evolved with a generator that recognizes the bootable JAR maven plugin.
+An https://github.com/eclipse/jkube/tree/v1.0.0-rc-1/quickstarts/maven/wildfly-jar[example] of Bootable JAR Maven plugin and JKube Maven Plugin to deploy
+application on OpenShift and Kubernetes. More information on the JKube Maven plugins can be found in JKube https://www.eclipse.org/jkube/docs/[documentation] 
+
+=== Nice-to-Have Requirements
+
+* Add the ability to copy content in the server during build. That is a common requirement (e.g.: auth properties files).
+* Ability to generate a runtime Maven repository in order to resolve the server artifacts from Maven local cache. 
+Huge benefit in term of JAR size and boot time (4/5 time faster, around 160ms to start vs 700 ms). 
+Specifically in a docker/openshift context, relying on slim server + Maven repository speeds up startup without impacting image size.
+Eg: java -Dmaven.repo.local=/maven-repo -jar myapp-bootable.jar
+
+
+=== Non-Requirements
+
+* Offer a new way to configure the server (e.g.: Thorntail yaml file).
```

Comment:
Thanks fixed.

Reply:
This would be a new RFE. CLI and standalone.xml file are the way to configure the server, defining a new syntax is out of scope. BTW, Galleon comes with its own syntax, a provisioning.xml can be used to mimic what we do with CLI. CLI is largely used and well-known. I would not drive users to use the galleon syntax that would implies a lot of new documentation.

Reply:
Thanks for the explanation.